packages/dev/gui/src/2D/controls/rectangle.ts

```diff
@@ -26,60 +24,74 @@ export class Rectangle extends Container {
         this._markAsDirty();
     }
 
-    /** Gets or sets the corner radius angle */
+    /** Gets or sets the corner radius of all angles */
     @serialize()
     public get cornerRadius(): number {
-        return this._cornerRadius;
-    }
-    public get cornerRadiusX(): number {
-        return this._cornerRadiusFree.x;
-    }
-    public get cornerRadiusY(): number {
-        return this._cornerRadiusFree.y;
-    }
-    public get cornerRadiusZ(): number {
-        return this._cornerRadiusFree.z;
-    }
-    public get cornerRadiusW(): number {
-        return this._cornerRadiusFree.w;
+        return this._cornerRadius[0];
     }
 
     public set cornerRadius(value: number) {
         if (value < 0) {
             value = 0;
         }
 
-        if (this._cornerRadius === value) {
+        if (this._cornerRadius[0] === value && this._cornerRadius[1] === value && this._cornerRadius[2] === value && this._cornerRadius[3] === value) {
             return;
         }
 
-        this._cornerRadius = value;
-        this._cornerRadiusFree.setAll(value);
+        this._cornerRadius = [value, value, value, value];

```

Comment:
This will create a new array and will cause GC trigger. If possible:

```suggestion
        this._cornerRadius[0] = his._cornerRadius[1] = his._cornerRadius[2] = his._cornerRadius[3] = value;

```